### PR TITLE
chore: remove old slots patch endpoint

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

we used to set up an echo.Any endpoint on `/agents` here we reduce that to just `echo.Get`
disallowing other methods.
this deprecates `PATCH /agents/:id/slots/:id`

- move cli to use the new apis for enabling and disabling slots

## Commentary (optional)

do we need to have an endpoint for setting the drain flag on slots?

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/cli/agent.py [hunk 1]
```diff
@@ -194,11 +194,9 @@ def patch(args: argparse.Namespace) -> None:
 def patch_slot(enabled: bool) -> Callable[[argparse.Namespace], None]:
     @authentication.required
     def patch(args: argparse.Namespace) -> None:
-        path = "agents/{}/slots/{}".format(args.agent_id, args.slot_id)
-        headers = {"Content-Type": "application/merge-patch+json"}
-        payload = {"enabled": enabled}
-
-        api.patch(args.master, path, json=payload, headers=headers)
+        bindings.post_EnableSlot(
```
Comment (hamidzr) [inline harness/determined/cli/agent.py:197]: wip

File: master/internal/api/error.go [hunk 2]
```diff
@@ -47,6 +47,10 @@ var (
 	ErrNotFound = errors.New("not found")
 	// ErrNotImplemented is the inner error for errors that convert to a 501.
 	ErrNotImplemented = errors.New("not implemented")
+
+	// ErrAPIRemoved is an error to inform the client they are calling an old, removed API.
+	ErrAPIRemoved = errors.New(`the API being called was removed,
+	please ensure the client consuming the API is up to date and report a bug if the problem persists`)
```
Comment (dzhu) [inline master/internal/api/error.go:53]: Remove extra spaces.

**Review by dzhu — APPROVED** [review]
This should be described as removing rather than deprecating, since the old API is no longer functional at all, even if the path does return some kind of response.

Also add a release note.

**Comment by hamidzr** [comment]
>This should be described as removing rather than deprecating, since the old API is no longer functional at all, even if the path does return some kind of response.

yep

**Pull request merged**